pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -25,7 +25,7 @@
   ninja,
 
   # dependencies for torch.utils.tensorboard
-  pillow, six, future, tensorflow-tensorboard, protobuf,
+  tensorboardSupport ? true, pillow, six, future, tensorflow-tensorboard, protobuf,
```

Comment:
This is mostly forbidden in pythonPackages because mixing different variants of the same package causes trouble in python.

Reply:
we should probably expose another `pytorchWithTensorboard` attr.

As long as it's the same underlying python package versions being exported, it should be okay if they get duplicated.

Reply:
I don't think adding a `tensorboardSupport` option is a good idea. Since, even if you disable this, PyTorch will still contain the bits that need TensorBoard.

Upstream PyTorch doesn't require TensorBoard. There is a PyTorch utility module that requires TensorBoard, but I think it's more intended for lazy loading (if you have TensorBoard installed, you can load the module).

I think if we want to be comprehensive about dependencies, we should always depend on TensorBoard. Or if we believe this is optional 'lazy loading' functionality (like upstream), we should completely remove these dependencies.

Reply:
Is there a better place for these kind of dependencies than `propagatedBuildInputs`? One issue with this `tensorboardSupport` option is that changing it forces a full rebuild even though nothing actually changes in this derivation -- it is only a runtime dependency of a specific submodule. In Debian terms, this would be a "Suggests" dependency (though the actual Debian `pytorch` package doesn't have tensorboard as a dependency at all). It would be ideal to have some way to not have these dependencies when they are not needed, like our case of building a minimal Docker image.

Anyway, would you prefer if I removed the "Make tensorboard dependencies optional" commit from this PR?

Reply:
I think we should seriously consider removing the Tensorboard dependency altogether. It is not necessary for normal PyTorch usage. I also know a lot of PyTorch users that use different means of tracking experiments (e.g. Wandb).

Another thing we could consider (I haven't looked at how feasible this is) is removing the tensorboard submodule altogether and moving it to a separate package. But we'd have to check whether this is possible while retaining the same import paths, etc.

Reply:
I removed the commit adding the `tensorboardSupport` option. I will open another PR for that, either splitting the `torch.utils.tensorboard` module into its own package if I can figure out how to do that, or else removing those dependencies.